src/components/VideoPlayer.vue

```diff
@@ -544,6 +537,20 @@ export default {
                         player.selectAudioLanguage(lang);
                     }
 
+                    (() => {
```

Comment:
Why are we creating a function here and call it immediately after?

Reply:
it's called [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE).
i used it to [Avoid polluting the global namespace](https://developer.mozilla.org/en-US/docs/Glossary/IIFE#avoid_polluting_the_global_namespace) and [Reduce Nesting, Reduce Complexity](https://testing.googleblog.com/2017/06/code-health-reduce-nesting-reduce.html).
in this case, to [reduce nested if](https://www.jetbrains.com/help/resharper/InvertIf.html)

without IIFE, the line `if (audioLanguages.length == 1 && audioLanguages[0] == "und") return;`  will exit code from function `setPlayerAttrs()` and will not execute the next code

without IIFE and attempt to reduce nesting, the code would look like this
```js
const audioLanguages = player.getAudioLanguages();
if (audioLanguages.length > 1) {
  const overflowMenuButtons = this.$ui.getConfiguration().overflowMenuButtons;
  // append language menu on index 1
  const newOverflowMenuButtons = [
      ...overflowMenuButtons.slice(0, 1),
      "language",
      ...overflowMenuButtons.slice(1),
  ];
  this.$ui.configure("overflowMenuButtons", newOverflowMenuButtons);
}
```

it's your call. i will make the change
i just want this feature..

Reply:
Thanks for the explanation!

In my opinion using an if statement looks simpler and more self-explaining here. "IIFE" is certainly a pretty cool thing, but only few people are familiar with and our goal is to make everything as easy to maintain and understand as possible.

Hence it would be great if you could replace the "IIFE" with an if loop (also in your other PR), then I'd be very glad to merge both of them (or Kavin) :)

Reply:
solved by https://github.com/TeamPiped/Piped/pull/2937/commits/9b740676acca541fc4432e932d02e047382b604b